### ISSUE TITLE
Accessibility for number spin boxes

### ISSUE THREAD

**Issue by shoogle** [musescore/MuseScore#13679]
**Describe the bug**
When a number spin box receives focus it is announced as a text edit field. Also, when using the up and down arrow keys to increment the number, most screen readers don't announce the changing value (only NVDA announces the change), and the editing caret disappears so it is no longer possible to type in the box.

**To Reproduce**
1. Open New Score dialog.
2. Pick any template and go to next screen (Additional score information).
3. Navigate to a spinbox on one of the popups (e.g. initial number of measures).
4. Press the up or down arrow key.
5. Press a number key on your keyboard.

**Expected behavior**
In step 3 the screen reader should say "spinner", "spin button", or similar (instead it says "edit" or "text edit").

In step 4 the screen reader should say the value of the number as it changes (NVDA does this but other screen readers do not).

After pressing the up or down arrow key, the digits in the number field should become selected (instead they become deselected and the editing caret disappears).

When you press a number key in step 5, it should replace the number in the box (instead nothing happens).

**Screenshots**
![image](https://user-images.githubusercontent.com/11011881/194780090-c440efe9-fe21-4d56-97c2-890773b06120.png)

**Platform information**
 - OS: Windows/macOS/Linux with a screen reader running

**Additional context**
Workaround: type the number manually with the keyboard without pressing the up or down arrow key first.

**Comment by cbjeukendrup** [comment]
I think this is a general issue because VoiceOver on macOS also treats spinboxes as text fields.

**Comment by shoogle** [comment]
I can confirm it affects Orca on Linux too, and I noticed a second part to the bug where incrementing the stepper causes the text caret to disappear so it is no longer possible to type in the box. I've changed the description accordingly and also upgraded the priority, though I think it is still appropriate to target this for post-beta.
